haigha/reader.py

```diff
@@ -232,7 +232,7 @@ def read_timestamp(self):
         Will raise BufferUnderflow if there's not enough bytes in the buffer.
         Will raise struct.error if the data is malformed
         """
-        return datetime.utcfromtimestamp(self.read_longlong())
+        return datetime.utcfromtimestamp(self.read_longlong() / 1e6)
```

Comment:
I found this error when using the RabbitMQ event exchange plugin. (Git address: https://github.com/rabbitmq/rabbitmq-event-exchange)
The timestamp is in microseconds and caused haigha failed to process the incoming message but I'm not sure whether it's the issue of haigha or the event exchange plugin.
So, I wonder if we could try to parse the timestamp in a more robust way: try the seconds and milliseconds and finally the microseconds, if none of these works, use None as the timestamp instead of raising an exception.
I'll send another pull request if necessary. thx. @vitaly-krugl